### DISCUSSION TITLE
Using Templater JS Files

### DISCUSSION THREAD

**Discussion by shabegom** [SilentVoid13/Templater#238] [Templates Showcase]
I wrote a tutorial on how to write user JS files and call them with a Templater Command:

https://publish.obsidian.md/shabegom/Publish/How+To+Use+Templater++JS+Scripts

**Comment by MaksimZinovev** [comment]
Thanks!

**Comment by ChristinWhite** [comment]
That was a great writeup and I definitely learned some things from the advanced section, thank you!

One thing that's been helping me out as I've been learning JS for Templater is [JSDoc](https://jsdoc.app) and the [VSCode type checking](https://code.visualstudio.com/Docs/languages/javascript#_jsdoc-support) to make sure I'm being consistent and providing intelligence about what I'm writing, do you know if it would be possible to import type definitions from Templater and the Obsidian API so things like `app.workspace.activeLeaf.view.editor` have some more information to work with, is that something that would be possible in a Templater user function?

**Reply by ChristinWhite** [reply]
I chatted with @shabegom about this on Discord and it sounds like there's not really a good way to do this.

**Comment by outdoorsman** [comment]
@shabegom Your tutorial link isn't working any more.

**Reply by martinellison** [reply]
still not working
